src/Framework/EnvironmentVariableReadEventArgs.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Arguments for the environment variable read event.
+    /// </summary>
+    [Serializable]
+    public class EnvironmentVariableReadEventArgs : BuildMessageEventArgs
```

Comment:
nit: maybe make the names uniform, either <Propertyoperation>Property or viceversa. For example: `UninitializedPropertyReadEventArgs`, `ReassignedPropertyEventArgs`, `EnvironmentVariableAddedAsPropertyEventArgs`

Reply:
The event we want is that an environment variable was read/used. Not that it was added as a property, right?

Reply:
Ah right, had gotten confused there :)